packages/framework/src/Markdown/Processing/DynamicMarkdownLinkProcessor.php

```diff
@@ -5,38 +5,68 @@
 namespace Hyde\Markdown\Processing;
 
 use Hyde\Hyde;
-use Hyde\Framework\Exceptions\RouteNotFoundException;
+use Illuminate\Support\Str;
+use Hyde\Support\Filesystem\MediaFile;
 use Hyde\Markdown\Contracts\MarkdownPostProcessorContract;
 
 class DynamicMarkdownLinkProcessor implements MarkdownPostProcessorContract
 {
     public static function postprocess(string $html): string
     {
-        foreach (static::patterns() as $pattern => $replacement) {
-            $html = preg_replace_callback($pattern, $replacement, $html);
+        foreach (static::routeMap() as $sourcePath => $route) {
+            $patterns = [
+                sprintf('<a href="%s">', $sourcePath),
+                sprintf('<a href="/%s">', $sourcePath),
+            ];
+
+            $html = str_replace($patterns, sprintf('<a href="%s">', $route->getLink()), $html);
+        }
+
+        foreach (static::assetMap() as $path => $mediaFile) {
+            $patterns = [
+                sprintf('<img src="%s"', $path),
+                sprintf('<img src="/%s"', $path),
+            ];
+
+            $html = str_replace($patterns, sprintf('<img src="%s"', static::assetPath($mediaFile)), $html);
         }
 
         return $html;
     }
 
-    /** @return array<string, callable(array<int, string>): string> */
-    protected static function patterns(): array
+    /**
+     * @return array<string, \Hyde\Support\Models\Route>
+     */
+    protected static function routeMap(): array
+    {
+        $map = [];
+
+        /** @var \Hyde\Support\Models\Route $route */
+        foreach (Hyde::routes() as $route) {
+            $map[$route->getSourcePath()] = $route;
+        }
+
+        return $map;
+    }
+
+    /**
+     * @todo Consider adding caching, since this is not a singleton like the routes.
```

Comment:
Yeah this needs caching. Benchmark shows a constant time of about `2ms` per iteration. 

```php
// Ensure the classes are loaded before the benchmark.
MediaFile::all();

\Illuminate\Support\Benchmark::dd(function () {
    MediaFile::all();
}, 1000); // "1.884ms", "1.884ms", "2.006ms"
```

I think we can use a simple static property/variable. In real usages this won't change during the application build cycle. In unit tests we can mock the method.

Reply:
Adding a static variable brings this so essentially zero (0.000ms)

```php
// Ensure the classes are loaded before the benchmark.
static::assetMap();

\Illuminate\Support\Benchmark::dd(function () {
    static::assetMap();
}, 100000); // 0.000ms
```